### PR TITLE
LPS-81968 Add CI test to check versions files

### PR DESCRIPTION
**Pull request by petershin**
Hi Andrea! Can you check if the generated files look ok?

If it does, please send this to Peter Yoo to check the test batch logic.

### CODE REVIEW

**Comment by petershin** [comment]
ci:test

File: build-test-batch.xml [hunk 1]
```diff
@@ -2981,6 +2981,56 @@ app.server.type=@{app.server.type}]]></echo>
 		<run-legacy-database-dump database.type="sybase" />
 	</target>
 
+	<target name="lib-version-builder-jdk8">
+		<run-batch-test>
+			<test-action>
+				<ant dir="portal-impl" target="build-lib-versions" />
+
+				<exec executable="git" output="${project.dir}/build.lib.versions.diff.txt">
```
Comment (Ithildir) [inline build-test-batch.xml]: Can you use `outputproperty` instead of writing into a temp file?

File: build-test-batch.xml [hunk 2]
```diff
@@ -2981,6 +2981,56 @@ app.server.type=@{app.server.type}]]></echo>
...
+			<test-set-up>
+				<if>
+					<isset property="env.DIST_PATH" />
+					<then>
+						<prepare-test-environment />
+					</then>
+					<else>
+						<ant target="compile" />
+
+						<ant target="install-portal-snapshots" />
```
Comment (Ithildir) [inline build-test-batch.xml]: I don't think you need this, do you?

File: build.properties [hunk 3]
```diff
@@ -204,6 +204,7 @@
     #    core,\
     #    util
 
+    build.lib.versions.skip.read.only=true
```
Comment (Ithildir) [inline build.properties]: You're skipping the read only repos so a merge pull from `com-liferay-apio-architect` with a new 3rd-party lib won't fail, right? But it's not nice to simply exclude read only repos 😞

I wonder if the merge pull could re-run `ant build-lib-versions` and commit...

Comment (petershin) [reply build.properties]: CI also copies over `private` modules from `master-private`, so the versions files need to be updated whenever the `git-commit` file is updated too.

I'll take a look and see if we can have a separate server automatically update the versions files once a day like the `Language.properties` translations.

Comment (Ithildir) [reply build.properties]: But we should probably only consider the public modules in `liferay-portal`, and both public and private in private branches of `liferay-portal-ee`.

**Comment by petershin** [comment]
ci:test

**Comment by petershin** [comment]
I'm looking into setting up separate server automatically update the `versions` files.  If its possible, then I think a separate job can be added to update the `SB` auto-generated files too

**Pull request closed without merging**